ml_peg/analysis/utils/utils.py

```diff
@@ -113,6 +114,54 @@ def rmse(ref: list, prediction: list) -> float:
     return mean_squared_error(ref, prediction)
 
 
+def build_density_inputs(
+    models: list[str],
+    model_stats: dict[str, dict[str, Any]],
+    property_key: str,
+    *,
+    metric_fn: Callable[[list, list], float] | None = None,
```

Comment:
I'm still a little confused by the purpose of this. This calculate the metric for the `model_stats`, but it's not currently used in the analysis script that calls this, which instead recalculates it?